src/NLog/Layouts/LayoutParser.cs

```diff
@@ -512,9 +521,17 @@ private static void MergeLiterals(List<LayoutRenderer> list)
             }
         }
 
-        private static LayoutRenderer ConvertToLiteral(LayoutRenderer renderer)
+        private static LayoutRenderer ConvertToLiteral(IRenderable renderer)
         {
-            return new LiteralLayoutRenderer(renderer.Render(LogEventInfo.CreateNullEvent()));
+            var logEventInfo = LogEventInfo.CreateNullEvent();
+            var text = renderer.Render(logEventInfo);
+            if (renderer is IRawValue rawValueRender)
+            {
+                var success = rawValueRender.TryGetRawValue(logEventInfo, out var rawValue);
```

Comment:
When not a success, then no need to change into `LiteralWithRawValueLayoutRenderer`

Reply:
Well less tricky to keep it like this.

Reply:
close as this is by design